nixos/modules/system/boot/networkd.nix

```diff
@@ -172,6 +172,14 @@ let
     (assertValueOneOf "AllSlavesActive" boolValues)
   ];
 
+  checkXfrm = checkUnitConfig "Xfrm" [
+    (assertOnlyFields [
+      "InterfaceId" "Independent"
```

Comment:
This looks okay according to the man pages usage of the Xfrm section:

```
[XFRM] SECTION OPTIONS
       The "[Xfrm]" section accepts the following keys:

       InterfaceId=
           Sets the ID/key of the xfrm interface which needs to be associated with a SA/policy. Can be decimal or hexadecimal, valid range is 0-0xffffffff,
           defaults to 0.

       Independent=
           Takes a boolean. If set to "no", the xfrm interface should have an underlying device which can be used for hardware offloading. Defaults to
           "no". See systemd.network(5) for how to configure the underlying device.

       For more detail information see Virtual xfrm interfaces[16]
```

This does not look complete when looking at the examples:

```      Example 17. /etc/systemd/network/27-xfrm.netdev

           [Xfrm]
           Name=xfrm0
           Kind=xfrm

           [Xfrm]
           Independent=yes
```

I guess that first `[Xfrm]` there should have have been `[NetDev]` then everything checks out.

I filed an upstream PR: https://github.com/systemd/systemd/pull/14630